extra_tests/snippets/stdlib_csv.py

```
@@ -181,6 +181,113 @@ def test_quote_minimal_writer_lineterminator():
+    class EmptyLineTerminator(csv.excel):
+        lineterminator = ""
+
+    keyword = io.StringIO()
+    csv.writer(keyword, lineterminator="").writerows([["a", "b"], ["c", "d"]])
+    assert keyword.getvalue() == "a,bc,d"
+
+    dialect = io.StringIO()
+    csv.writer(dialect, dialect=EmptyLineTerminator).writerows([["a", "b"], ["c", "d"]])
+    assert dialect.getvalue() == "a,bc,d"
+
+    input = "a,b\r\nc,d\n"
```

Ruff (0.15.21)

[error] Variable `input` is shadowing a Python builtin

(A001)